editoast/src/models/work_schedules.rs

```diff
@@ -38,3 +45,45 @@ pub struct WorkSchedule {
     pub work_schedule_type: WorkScheduleType,
     pub work_schedule_group_id: i64,
 }
+
+impl WorkSchedule {
+    fn map_to_core_work_schedule(
+        &self,
+        start_time: DateTime<Utc>,
+    ) -> crate::core::stdcm::WorkSchedule {
+        crate::core::stdcm::WorkSchedule {
+            start_time: elapsed_since_time_ms(&self.start_date_time, &start_time),
+            end_time: elapsed_since_time_ms(&self.end_date_time, &start_time),
+            track_ranges: self
+                .track_ranges
+                .iter()
+                .map(|track| UndirectedTrackRange {
+                    track_section: track.track.to_string(),
+                    begin: (track.begin * 1000.0) as u64,
+                    end: (track.end * 1000.0) as u64,
+                })
+                .collect(),
+        }
+    }
+
+    pub fn get_id_and_core_work_schedule(
+        &self,
+        start_time: DateTime<Utc>,
+        latest_simulation_end: DateTime<Utc>,
+    ) -> Option<(i64, crate::core::stdcm::WorkSchedule)> {
+        let search_window_duration = (latest_simulation_end - start_time).num_milliseconds() as u64;
+
+        let ws = self.map_to_core_work_schedule(start_time);
+        if ws.end_time > 0 && ws.start_time < search_window_duration {
+            return Some((self.id, ws));
+        }
+        None
+    }
+}
+
+fn elapsed_since_time_ms(time: &NaiveDateTime, start_time: &DateTime<Utc>) -> u64 {
```

Comment:
big nit, feel free to ignore:

```suggestion
fn elapsed_time_since_ms(time: &NaiveDateTime, since: &DateTime<Utc>) -> u64 {
```

Also can't we have `time: &Datetime<Utc>` for consistency?

Reply:
Done.